### PR TITLE
Change proc path

### PR DESCRIPTION
**Pull request by amitlimaye**
Change proc path to a value that can be passed to newenforcer and also add a check to ensure that we don't launch a enforcer in a host network namespace

### CODE REVIEW

**Comment by bvandewalle** [comment]
When creating the enforcer proxy// Remote enforcer, the new proc path should be an optional parameter, so that It can be defined on startup.

File: cmd/remoteenforcer/remoteenforcer_linux.go [hunk 1]
```diff
@@ -28,24 +25,25 @@ import (
 )
 
 const (
-	envSocketPath = "SOCKET_PATH"
-	envSecret     = "SECRET"
-	nsErrorState  = "NSENTER_ERROR_STATE"
-	nsEnterLogs   = "NSENTER_LOGS"
+	envSocketPath     = "SOCKET_PATH"
```
Comment (satyamsi) [inline cmd/remoteenforcer/remoteenforcer_linux.go]: nice-to-have: prefix "APORETO_" to all env variables

File: cmd/remoteenforcer/remoteenforcer_linux.go [hunk 2]
```diff
@@ -94,34 +95,9 @@ func (s *Server) InitEnforcer(req rpcwrapper.Request, resp *rpcwrapper.Response)
 		return errors.New(resp.Status)
 	}
 
-	pid := strconv.Itoa(os.Getpid())
```
Comment (satyamsi) [inline cmd/remoteenforcer/remoteenforcer_linux.go]: Resurrect this when not doing private mount as this is the more common code path and this check helps.

File: configurator/configurator.go [hunk 3]
```diff
@@ -44,6 +44,7 @@ func NewTriremeLinuxProcess(
 			nil,
 			secrets,
 			constants.LocalServer,
+			"/proc",
```
Comment (satyamsi) [inline configurator/configurator.go]: constant def

File: enforcer/utils/nsenter/nsenter.c [hunk 4]
```diff
@@ -15,15 +15,21 @@ void nsexec(void) {
...
+  char mountpoint[STRBUF_SIZE] = {0};
+  char *str = getenv("CONTAINER_PID");
+  char *proc_mountpoint = getenv("PROC_MOUNTPOINT");
   if(str == NULL){
     // We are not running as remote enforcer
     setenv("NSENTER_LOGS", "no container pid", 1);
     return;
   }
-
+  if(proc_mountpoint == NULL){
```
Comment (satyamsi) [inline enforcer/utils/nsenter/nsenter.c]: spaces between commas

File: processmon/processmon.go [hunk 5]
```diff
@@ -170,14 +171,40 @@ func (p *ProcessMon) KillProcess(contextID string) {
...
+func (p *ProcessMon) LaunchProcess(contextID string, refPid int, rpchdl rpcwrapper.RPCClient, arg string, statsServerSecret string, procMountPoint string) error {
 	secretLength := 32
 	var cmdName string
 
 	_, err := p.activeProcesses.Get(contextID)
 	if err == nil {
 		return nil
 	}
+
+	pidstat, pidstaterr := os.Stat("/proc/" + strconv.Itoa(refPid) + "/ns/net")
```
Comment (satyamsi) [inline processmon/processmon.go]: mountPoint instead of '/proc'

File: enforcer/datapath_test.go [hunk 6]
```diff
@@ -59,7 +59,7 @@ func TestInvalidContext(t *testing.T) {
 
 		secret := tokens.NewPSKSecrets([]byte("Dummy Test Password"))
 		collector := &collector.DefaultCollector{}
-		enforcer := NewDefaultDatapathEnforcer("SomeServerId", collector, nil, secret, constants.LocalContainer).(*datapathEnforcer)
+		enforcer := NewDefaultDatapathEnforcer("SomeServerId", collector, nil, secret, constants.LocalContainer, "/proc").(*datapathEnforcer)
```
Comment (satyamsi) [inline enforcer/datapath_test.go]: /proc as constant

File: monitor/linuxmonitor/cgnetcls/cgnetcls.go [hunk 7]
```diff
@@ -175,6 +176,31 @@ func (s *netCls) Deletebasepath(cgroupName string) bool {
 	return false
 }
 
+func mountcgroupcontroller() {
```
Comment (satyamsi) [inline monitor/linuxmonitor/cgnetcls/cgnetcls.go]: CamelCase

File: cmd/remoteenforcer/remoteenforcer_linux.go [hunk 8]
```diff
@@ -55,14 +56,17 @@ func NewServer(service enforcer.PacketProcessor, rpchdl rpcwrapper.RPCServer, rp
 	if err != nil {
 		return nil, err
 	}
-
+	procMountPoint := os.Getenv(envProcMountPoint)
+	if len(procMountPoint) == 0 {
+		procMountPoint = "/proc"
```
Comment (satyamsi) [inline cmd/remoteenforcer/remoteenforcer_linux.go]: Why isnt /proc a const ?

File: configurator/configurator.go [hunk 9]
```diff
@@ -23,6 +23,10 @@ import (
 	"github.com/aporeto-inc/trireme/supervisor/proxy"
 )
 
+const (
+	defaultProcMountPoint = "/proc"
+)
```
Comment (satyamsi) [inline configurator/configurator.go]: Add AporetoProcMountPoint and change default to Default.

**Pull request merged**